core/src/main/java/io/paradoxical/cassieq/admin/resources/api/v1/AccountResource.java

```diff
@@ -74,15 +121,145 @@ public Response getAccount(
 
             final Optional<AccountDefinition> account = accountRepository.getAccount(accountName);
 
-            if(!account.isPresent()){
+            if (!account.isPresent()) {
                 return Response.status(Response.Status.NOT_FOUND).build();
             }
 
             return Response.ok(account.get()).build();
         }
         catch (Exception e) {
-            logger.error(e, "Error");
-            return buildErrorResponse("CreateQueue", null, e);
+            logger.with("account-name", accountName).error(e, "Error getting account");
+
+            return buildErrorResponse("GetAccount", null, e);
+        }
+    }
+
+    @DELETE
+    @Timed
+    @Path("/{accountName}/keys/{keyName}")
+    @ApiOperation(value = "Delete Account Key")
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "Ok"),
+                            @ApiResponse(code = 500, message = "Server Error") })
+    public Response deleteAccountKey(@PathParam("accountName") AccountName accountName, @PathParam("keyName") KeyName keyName) {
+        try {
+            final AccountRepository accountRepository = dataContextFactory.getAccountRepository();
+
+            final Optional<AccountDefinition> account = accountRepository.getAccount(accountName);
+
+            if (!account.isPresent()) {
+                return Response.status(Response.Status.NOT_FOUND).build();
+            }
+
+            final AccountDefinition accountDefinition = account.get();
+
+            final HashMap<KeyName, AccountKey> prunedKeys = new HashMap<>(accountDefinition.getKeys());
+
+            prunedKeys.remove(keyName);
+
+            accountDefinition.setKeys(ImmutableMap.copyOf(prunedKeys));
+
+            accountRepository.updateAccount(accountDefinition);
+
+            return Response.ok(accountDefinition).build();
+
+        }
+        catch (Exception e) {
+            logger.with("account-name", accountName)
+                  .with("key-name", keyName)
+                  .error(e, "Error getting account");
+
+            return buildErrorResponse("RemoveAccountKey", null, e);
+        }
+    }
+
+    @POST
+    @Timed
+    @Path("/{accountName}/keys/{keyName}")
```

Comment:
I'm fine with that, I was taking a shortcut. A payload is nicer cause if we want to let them set the private key blob they can also do that and we can take it